### PR TITLE
Add {In,Out}_channel.isatty

### PR DESCRIPTION
**Pull request by nojb**
Simply expose the existing primitive in `Sys` (the same code used to implement `Unix.isatty`). Motivated by discussion in #11127. cc @dbuenzli

### CODE REVIEW

**Comment by dbuenzli** [comment]
Wonderful! 

But now that I see the signature, my idea to put in in `Sys` was misleading. `isatty` is can be applied to both input and output fds. 

That looks an excellent opportunity to use the new `{In,Out}_channel` modules.

**Comment by nojb** [comment]
> Wonderful!
> 
> But now that I see the signature, my idea to put in in `Sys` was misleading. `isatty` is can be applied to both input and output fds.
> 
> That looks an excellent opportunity to use the new `{In,Out}_channel` modules.

Good point, amended.

File: stdlib/out_channel.mli
Comment (dbuenzli) [inline stdlib/out_channel.mli]: You could have retained the documentation style of `is_buffered` (i.e. `isatty oc` is `true` if the channel `oc` refers…). But I'm nitpicking.

**Review by dbuenzli — APPROVED** [review]

File: stdlib/out_channel.mli
Comment (nojb) [reply stdlib/out_channel.mli]: Wish granted, thanks.

**Review by gasche — APPROVED** [review]

File: stdlib/in_channel.mli
Comment (avsm) [inline stdlib/in_channel.mli:165]: A nitpick is that the [Windows documentation for isatty](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/isatty?view=msvc-170) refers to:
>  a character device (a terminal, console, printer, or serial port).

Perhaps it's more accurate to refer to a character device instead, but I haven't checked to see what isatty on Linux does does if you give it a serial port device.

Comment (nojb) [reply stdlib/in_channel.mli:165]: > I haven't checked to see what isatty on Linux does does if you give it a serial port device.

It returns `false`. Personally I would keep the existing somewhat vague docstring.

**Comment by nojb** [comment]
Thanks @gasche for the approval. Per the rule, this PR needs a second approval from a core dev before it can be merged, any takers?

**Comment by gasche** [comment]
Note: I wondered how this new feature of channels would be adapted for the proposal of making the channel type externally-extensible ([RFC: modular IO](https://github.com/ocaml/RFCs/pull/19)). My conclusion was that it adds yet another small wrinkle to the proposal, but nothing worse than what we already have with, say, `fseek`, so it's probably fine.

**Comment by nojb** [comment]
Friendly ping (still in need of a second approval for this easy PR). Maybe @Octachron?

**Comment by Octachron** [comment]
I might have some time once the stdlib audit and concurrency-safety documentation for 5.0 is done.

**Comment by dbuenzli** [comment]
I'm sure @Octachron has enough things on his plate. But could we maybe get approval from a less busy core dev team member to get this in ?

**Comment by gasche** [comment]
I opened the thread to join to help, and then I realized that I am the first approver...

**Review by Octachron — APPROVED** [review]
The `isatty` function seems like a fine addition to me:
- the function already exists and is merely exposed
- it is an important property of the underlying channel
- it should relatively straightforward for custom channels to implement this function

**Comment by nojb** [comment]
> The `isatty` function seems like a fine addition to me:

Thanks for the approval. I updated the "since" docstring to `5.1.0` and moved the Changes entry to the corresponding section. Planning to merge once CI passes.

**Pull request merged**
